bridge/bridge/deploy/local.py

```diff
@@ -0,0 +1,283 @@
+
+        if port is None:
+            return 404
+
+        resp = requests.request(
+            method=request.method,
+            url=f"http://localhost:{port}/{path}",
+            headers={
+                key: value for (key, value) in request.headers if key != "Host"
+            },
+            data=request.get_data(),
+            cookies=request.cookies,
+            allow_redirects=False,
+            params=dict(request.args),
+        )
+
+        excluded_headers = [
+            "content-encoding",
+            "content-length",
+            "transfer-encoding",
+            "connection",
+        ]
+        headers = [
+            (name, value)
+            for (name, value) in resp.raw.headers.items()
+            if name.lower() not in excluded_headers
+        ]
+
+        response = Response(resp.content, resp.status_code, headers)
+        return response
+
+    def run(self):
+        port = os.environ.get("PORT", 3000)
+        self.handle = threading.Thread(
+            target=self.app.run, kwargs={"host": "0.0.0.0", "port": port}
+        )
+        self.handle.start()
+
+
+def destroy_failed_conda(model_dir):
+    from mlflow.utils.conda import _get_conda_env_name  # type: ignore
+    from mlflow.pyfunc import _load_model_env  # type: ignore
+    from shutil import rmtree
+
+    env = _load_model_env(model_dir)
+    conda_env_path = f"{model_dir}/{env}"
+    logger.info(f"Conda env path: {conda_env_path}")
+    name = _get_conda_env_name(conda_env_path)
+    logger.info(f"Destroying Conda environment {name}.")
+    handle = Popen(
+        [
+            "conda",
+            "env",
+            "remove",
+            "-n",
+            name,
+        ],
+    )
+    handle.wait()
+    prefix = os.environ.get("CONDA_PREFIX", "/opt/conda")
+    conda_env_dir = f"{prefix}/envs/{name}"
+    logger.info(f"Conda Env Dir: {conda_env_dir}")
+    rmtree(conda_env_dir, ignore_errors=True)
+    logger.info(f"Done {handle.returncode}")
+
+
+def prepare_conda_environment(model_dir, returns):
+    from filelock import FileLock
+    from mlflow.utils.conda import get_or_create_conda_env  # type: ignore
+    from mlflow.pyfunc import _load_model_env  # type: ignore
+
+    env = _load_model_env(model_dir)
+    conda_env_path = f"{model_dir}/{env}"
+    lock = FileLock(CONDA_LOCK_FILE)
+    logger.info(f"Obtaining lock {CONDA_LOCK_FILE}.")
+    try:
+        with lock:
+            logger.info("Got lock.")
+            get_or_create_conda_env(conda_env_path)
+        returns[0] = 0
+    except Exception as e:
+        logger.exception(e)
+        returns[0] = 1
+
+
+class ModelDeployment:
+    def __init__(self, model_version: ModelVersion, model_uri: str, port: int):
+        self.model_version = model_version
+        self.model_uri = model_uri
+        self.serve_handle = None
+        self.port = port
+        self._install()
+
+    def _install(self):
+        # Install return code
+        self.returns = [None]
```

Comment:
nit: for consistency this should be set in the init like `serve_handle` or `serve_handle` should be set in `serve`

Reply:
I'm not sure about consistency, but this is set right before the thread is dispatched which will write the results back to this variable. I think it makes sense to (re)set it here. 

`serve_handle` is set in `__init__`, because the first call to `ModelDeployment.check` will try to read it on  line 151, before `ModelDeployment._serve` is called.

Reply:
I see